File: sub-mysql-web-service/src/main/java/org/github/mysql/web/shiro/service/impl/ShiroServiceImpl.java
package org.github.mysql.web.shiro.service.impl;

import lombok.NonNull;
import lombok.val;
import org.apache.shiro.authc.AuthenticationException;
import org.github.mysql.web.shiro.mapper.IShiroMapper;
import org.github.mysql.web.shiro.model.UserDTO;
import org.github.shiro.AuthorInfo;
import org.github.shiro.IShiroService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

import static org.github.mysql.web.base.enums.Deleted.deleted;
import static org.github.mysql.web.base.enums.Enable.disabled;

@Service
public class ShiroServiceImpl implements IShiroService {
  @Autowired
  private IShiroMapper shiroMapper;

  @Override
  public @NonNull List<AuthorInfo> queryAuthorInfo(@NonNull String userId) {
    return shiroMapper.queryAuthorInfo(Long.valueOf(userId));
  }

  @Override
  public @NonNull UserDTO queryUser(@NonNull String username) {
    val user = shiroMapper.queryUserInfo(username);
    if (user == null) throw new AuthenticationException("用户未注册");
    if (disabled.getCode().equals(user.getStatus())) throw new AuthenticationException("该用户已禁用");
    if (deleted.getCode().equals(user.getDeleted())) throw new AuthenticationException("该用户已注销");
    return user;
  }
}

File: sub-mysql-mydb-service/src/main/java/org/github/mysql/mydb/auth/service/impl/ShiroServiceImpl.kt
package org.github.mysql.mydb.auth.service.impl

import org.apache.shiro.authc.AuthenticationException
import org.github.mysql.mydb.auth.mapper.IShiroMapper
import org.github.mysql.mydb.auth.service.IShiroService
import org.github.mysql.mydb.base.dict.Deleted.DELETED
import org.github.mysql.mydb.base.dict.Enabled.DISABLED
import org.github.shiro.AuthorInfo
import org.github.shiro.User
import org.github.spring.restful.json.JSONArrayReturn
import org.github.spring.restful.json.JSONDataReturn
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.Primary
import org.springframework.stereotype.Service

@Primary
@Service
class ShiroServiceImpl: IShiroService {
  @Autowired
  private lateinit var shiroMapper: IShiroMapper

  override fun queryAuthorInfo(userId: String): JSONArrayReturn<AuthorInfo> {
    val list = shiroMapper.queryAuthorInfo(userId.toLong())
    return JSONArrayReturn.of(list)
  }

  override fun queryUser(username: String): JSONDataReturn<out User> {
    val user = shiroMapper.queryUserInfo(username) ?: throw AuthenticationException("用户未注册")
    if (DISABLED.code == user.enabled) throw AuthenticationException("该用户已禁用")
    if (DELETED.code == user.deleted) throw AuthenticationException("该用户已注销")
    return JSONDataReturn.of(user)
  }
}
